### DISCUSSION TITLE
How to make sure a distant referenced schema version will not change with time ?

### DISCUSSION THREAD

**Discussion by seedy** [schemaorg/schemaorg#2779] [Q&A]
Hey there :wave: 

### Context

I am working for [Misakey](https://github.com/misakey), a company handling end-to-end data encryption and decryption for user data.

#### Interoperable data

We plan on drafting features to handle interoperable encrypted data between different actors - for instance service companies - at user's decision.

I recently started digging into JSON-LD and schema.org as a data type provider.

To achieve our goal, we would need to define schemas for data providers which could be used to encrypt some incoming data according to that said schema.

We must make sure decryption at any other moment would be done with the same schema regardless of possible updates applied in the meantime.

This brings up questions regarding schema versioning.

### Schema versioning

I guess you already experienced breaking changes in schema versions.

How do you handle so far such changes ? Is there a recommended pattern to define versioned datatype schemas ?

To me, a simple and straightforward solution would be to add a `version` key to all our data types.

But, this means we cannot use already existing schemas from other providers unless they also follow our pattern. That's why it seems a bit clunky in my opinion.

**Comment by RichardWallis** [comment]
Although not from the Schema.org site itself (which only serves the latest version), the [download files defining the vocabulary](https://schema.org/docs/developers.html#defs) are archived in release versioned directories in the Github repository.

For example: https://github.com/schemaorg/schemaorg/tree/main/data/releases/11.0

**Comment by chrisspradling1980** [comment]
PERFECT!!!!!!!!!!!

**Comment by danbri** [comment]
We also have a http://schema.org/schemaVersion property for the rare cases when a document wants to commit more explicitly to a specific version of schema.org.
